python/social_bot/envs/grocery_ground.py

```diff
@@ -191,20 +207,49 @@ def run(self, agent, world):
             agent (pygazebo.Agent): the learning agent 
             world (pygazebo.World): the simulation world
         """
-        pre_agent_pos = self.task_specific_observation()[:2]
+        self._pre_agent_pos = self.get_icub_extra_obs(agent)[:3]
         agent_sentence = yield
         done = False
+        # set icub initial pose
+        x = self._agent_init_pos[0] + random.random() * self._random_range
+        y = self._agent_init_pos[1] + random.random() * self._random_range
+        agent_pos = np.array([x, y, 0.6])
+        # a trick from roboschool, to encourage optimal straight walk
+        orient = 0
+        if random.randint(0, 2) == 0:
+            orient = self._get_angle_to_target(agent_pos, 'iCub::root_link',
+                                               np.pi)
+        pose = (agent_pos, np.array([0, 0, orient]))
+        # one problem here is we can not control the sequence of task run(), if 
```

Comment:
one problem here is we can not control the sequence of task run(), if this aux task is latter than main task, the target might be at initial pose and has not been randomly moved yet.

Reply:
Right now, teacher go through the task groups in the same order as they are added to teacher. So it should not be a problem.